### DISCUSSION TITLE
Some build comments

### DISCUSSION THREAD

**Discussion by veduco** [dd010101/vyos-jenkins#3] [General]
> iproute2 (unknown)

The package included in the repo is 6.6. My 1.4 build pulls in `iproute2	6.8.0-1~bpo12+1`.
You can see the commits that changed and pulled iproute from the build, allowing the package to be sourced from debian:
```
commit 8c829af7b643cb4a8626e21438c45fe3126d64ec
Merge: cfe3a13 fa5d878
Author: Christian Breunig <christian@breunig.cc>
Date:   Thu Feb 29 21:43:21 2024 +0100

    Merge pull request #520 from vyos/mergify/bp/sagitta/pr-519
    
    Revert "iproute2: T6078: upgrade package to 6.6 to match Kernel" (backport #519)

commit fa5d87896674d8887432e7cfeccd9122ee5649db
Author: Christian Breunig <christian@breunig.cc>
Date:   Thu Feb 29 21:40:01 2024 +0100

    Revert "iproute2: T6078: upgrade package to 6.6 to match Kernel"
    
    This reverts commit 60d16e494958bfe4a43048ee3a7f63fd57600875.
    
    We already use iproute2 6.7 from Debian Bookworm Backports
    
    (cherry picked from commit c56d76986f9efa05c4348401e7cdd05073aed2e7)

commit 9cf88379d1657fd75d9ccd7c9076bca898c21f18
Merge: 5bb75ef 40d8231
Author: Christian Breunig <christian@breunig.cc>
Date:   Thu Feb 29 11:51:45 2024 +0100

    Merge pull request #515 from vyos/mergify/bp/sagitta/pr-514
    
    T6078: package upgrade to v6.6 to match kernel for ethtool and iproute2 (backport #514)

commit 40d8231525e84b840ac1163cbbb1e66bb71c15af
Author: Christian Breunig <christian@breunig.cc>
Date:   Wed Feb 28 21:21:57 2024 +0100

    iproute2: T6078: upgrade package to 6.6 to match Kernel
    
    (cherry picked from commit 60d16e494958bfe4a43048ee3a7f63fd57600875)
```
-------
> libnss-tacplus + libtacplus-map (unknown)

This package must be build using the vyos sourced repos:
`libtacplus-map`, `libpam-tacplus`, and then if those are succesful, `libnss-tacplus`

In my experience, building with `pam_tacplus` from the `packages/` directory yields a 1.7.0 version whereas the repo lists much lower versions provided by Cumulus Linux.

`libnss-tacplus` has dependencies that are not met by the docker build:
```
autoconf-archive
libaudit-dev
```

`libpam-tacplus` has dependencies that are not met by the docker build:
```
libpam-dev
autoconf-archive
libaudit-dev
```

Once those two builds are successful, you can build `libnss-tacplus` by installing the generated DEBs from the above -tacplus steps, and then installing the dependencies that are not met by the docker build:
```
libaudit-dev
```
You then have to disable the Werror compile check:
```
sed 's/if test "x$GCC" = "xyes"; then/if false; then/' -i configure.ac
```
and then run 
```
dpkg-buildpackage -uc -us -tc -b
```

Yields:

```
libnss-tacplus-dbgsym_1.0.4-cl5.1.0u11_amd64.deb
libnss-tacplus_1.0.4-cl5.1.0u11.dsc             
libnss-tacplus_1.0.4-cl5.1.0u11.tar.gz          
libnss-tacplus_1.0.4-cl5.1.0u11_amd64.buildinfo 
libnss-tacplus_1.0.4-cl5.1.0u11_amd64.changes   
libnss-tacplus_1.0.4-cl5.1.0u11_amd64.deb       
libpam-tacplus-dbgsym_1.4.3-cl5.1.0u5_amd64.deb 
libpam-tacplus-dev_1.4.3-cl5.1.0u5_amd64.deb    
libpam-tacplus_1.4.3-cl5.1.0u5.dsc              
libpam-tacplus_1.4.3-cl5.1.0u5.tar.gz           
libpam-tacplus_1.4.3-cl5.1.0u5_amd64.buildinfo  
libpam-tacplus_1.4.3-cl5.1.0u5_amd64.changes    
libpam-tacplus_1.4.3-cl5.1.0u5_amd64.deb        
libtacplus-map-dev_1.0.1-cl5.1.0u9_amd64.deb    
libtacplus-map1-dbgsym_1.0.1-cl5.1.0u9_amd64.deb
libtacplus-map1_1.0.1-cl5.1.0u9_amd64.deb       
libtacplus-map_1.0.1-cl5.1.0u9.dsc              
libtacplus-map_1.0.1-cl5.1.0u9.tar.gz           
libtacplus-map_1.0.1-cl5.1.0u9_amd64.buildinfo  
libtacplus-map_1.0.1-cl5.1.0u9_amd64.changes    

and I believe these as well:

libtac-dev_1.4.3-cl5.1.0u5_amd64.deb            
libtac2-bin-dbgsym_1.4.3-cl5.1.0u5_amd64.deb    
libtac2-bin_1.4.3-cl5.1.0u5_amd64.deb           
libtac2-dbgsym_1.4.3-cl5.1.0u5_amd64.deb        
libtac2_1.4.3-cl5.1.0u5_amd64.deb               
```
-------

> python3-vici

Similar to this documentation here: https://docs.vyos.io/en/equuleus/contributing/upstream-packages.html#strongswan

You have to pass `--enable-python-eggs` to the build. The easiest way seems to be injecting the necessary line into one of the to-be-inserted patches. Code provided by another gentleman.
```
sed -i 's/dpkg-buildpackage -uc -us -tc -b -d//' build.sh
sed -i 's/+CONFIGUREARGS_VYOS := --disable-warnings/+CONFIGUREARGS_VYOS := --disable-warnings --enable-python-eggs/' patches/0004-VyOS-disable-options-enabled-by-Debian-that-are-unus.patch
./build.sh
```
Yields:
```
python3-vici_5.9.11-1_all.deb
```

------
> vpp 

VPP is a work in progress by the VyOS team, and should they succeed in building VPP into the product, it will not be a free add-on. The VPP package can be sourced straight from upstream: https://s3-docs.fd.io/vpp/24.06/gettingstarted/installing/ubuntu.html

https://forum.vyos.io/t/vpp-addon-business-model-still-open-source/13177

------
> amazon-cloudwatch-agent

Source is here: https://github.com/aws/amazon-cloudwatch-agent
It can be built, but is only used in the AWS ISO build as evidenced by the build flavor: https://github.com/vyos/vyos-build/blob/sagitta/data/build-flavors/aws-iso.toml

----
> Equuleus xe-guest-utilities

I too noted the addition of the `xe-guest-utilities` package. I suspect it comes from upstream: https://github.com/xenserver/xe-guest-utilities

That said, the vyox-xe-guest-utilities package is what gets pulled in and allows my builds to succeed.

-------

Now, it was stated that everything required to build is located in the repos, and that seems to be 95% true, but the above sure seems to indicate that manual intervention is required to get a repo initially seeded.

Thank you for your work on this.

**Comment by dd010101** [comment]
Thank you as well. Now I should have enough information.

I don't get it the VPP, it was used but it's not used anymore? It was included in current but they dropped it later?

**Reply by dd010101** [reply]
I was lead into thinking that everything is built from vyos repositories since that's what the vyos team said, yet it's not. They don't know what they have or they didn't expect anyone would actually try to do it?

**Reply by veduco** [reply]
I would suspect that VPP was just added to their repo so that they didn't have to execute the additional step of adding the VPP repo during development, as well as effectively pinning the version. Let's not forget that 1.4 was 'current' until September 25, 2023: https://github.com/vyos/vyos-build/commit/1746109255d3d4cdde3540f8eb099adddccd2468. VPP in sagitta is dated September 26, 2023. VPP in current is dated April 23, 2024. This likely indicates it's not a maintained package for sagitta.

**Reply by veduco** [reply]
I would also suspect that the state of the repos is simply a rolling forward copy of previous branches, i.e. to create the `current` repo, you'd create a directory for `equuleus` (or older) packages, and then build or add `sagitta` packages into that directory, and then build or add `current` packages to that directory. And then when `current` is renamed `circinus`, rinse and repeat.
